cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -238,6 +238,34 @@ TEST(ExecPlanExecution, SourceSink) {
   }
 }
 
+TEST(ExecPlanExecution, TableSourceSink) {
+  for (bool slow : {false, true}) {
+    SCOPED_TRACE(slow ? "slowed" : "unslowed");
+
+    for (bool parallel : {false, true}) {
+      SCOPED_TRACE(parallel ? "parallel" : "single threaded");
+
+      ASSERT_OK_AND_ASSIGN(auto plan, ExecPlan::Make());
+      AsyncGenerator<util::optional<ExecBatch>> sink_gen;
+
+      auto exp_batches = MakeBasicBatches();
+      ASSERT_OK_AND_ASSIGN(auto table,
+                           TableFromExecBatches(exp_batches.schema, exp_batches.batches));
+
+      ASSERT_OK(
+          Declaration::Sequence({
+                                    {"table_source", TableSourceNodeOptions{table, 1}},
```

Comment:
Can you test two different batch sizes?  I like 1 as a test case but can you also test 4 (there are no batches of size 4 in the table but this should still be a valid setting, it won't concatenate batches and will just emit the undersized batches which is ok).

Reply:
Yes, it is good to have that. Let me add it.